Docs: Add an examples section (#9371)
readthedocs/readthedocs.org: docs/user/examples.rst

> Example projects

Author: > At least a bit more copy explaining how to use the example projects (do I import them into RTD? Fork them?) is probably important.

This is now done! Added a sentence more with some over-all guidance.

I've added another h2 "Sphinx and MkDocs examples" which will be the first major topic headline, to be adapted as we have more tables of examples.

I need a bit more enlightenment on how you see SEO as a goal in itself on a content-based site like this. Shouldn't it ideally be that SEO naturally follows from having made a logical content structure with great contents?

> Example projects

Reviewer: SEO doesn't follow naturally if we don't use keywords that users will search for. Having good keywords in our h1/h2 is quite important for users to be able to find what they want.

The following example projects show a rich variety of uses of Read the Docs. You can use them for inspiration, for learning and for recipies to start your own documentation projects.

.. list-table::
   :header-rows: 1
   :widths: 20 20 40 20

   * - Topic
     - Framework
     - Description
     - Links
   * - Sphinx
     - Sphinx
     - Basic example of using Sphinx
     - `[Git] <https://github.com/readthedocs-examples/example-sphinx-basic/>`__ `[Rendered] <https://example-sphinx-basic.readthedocs.io/en/latest/>`__
   * - MkDocs
     - MkDocs
     - Basic example of using MkDocs
     - `[Git] <https://github.com/readthedocs-examples/example-mkdocs-basic/>`__ `[Rendered] <https://example-mkdocs-basic.readthedocs.io/en/latest/>`__

Real-life examples
------------------

.. image:: _static/images/awesome-list.svg
  :alt: Awesome List badge
  :target: https://github.com/readthedocs-examples/

We maintain an **Awesome List** where you can contribute new shiny examples of using Read the Docs. Please refer to the instructions on how to submit new entries on `Awesome Read the Docs Projects <https://github.com/readthedocs-examples/>`_.

New example projects
--------------------

> We would love to add more examples that showcase documentation options. We don't want your example project to break quickly, therefore we require that your example project...

Reviewer: ```suggestion
We would love to add more examples that showcase feature of Read the Docs, or great documentation projects.
We want to ensure these are useful for other community members, so we have the following requirements for building an example project:
```

Just a bit more friendly language (s/break quickly/be maintainable/) :)

> * is **hosted and maintained *by you*** in its own Git repository, ``example-<topic>``

Reviewer: ```suggestion
* is **hosted and maintained by you** in its own Git repository, ``example-<topic>``
```

What does `by you` mean here? We should either remove it or explain it.

Reviewer: This was resolved, but the issue wasn't addressed.

* uses a ``.readthedocs.yaml`` configuration
* has **continuous integration and PR builds**

> * is **versioned** as a real software project

Reviewer: Maybe this is a recommendation instead of a requirement? Seems like a lot for an example project, and probably not obvious how it would work in general.

Author: Yes, agreed -- both the versioning and CI requirements are moved so they appear as recommendations.

* is finally added to the above list by **opening a PR** targeting `examples.rst <https://github.com/readthedocs/readthedocs.org/blob/main/docs/user/examples.rst>`_

We **recommend** that your project...

* covers basic features, but references real-life projects whenever possible
* has a minimal tech stack -- or whatever you feel comfortable about maintaining

> * copies from an existing example project as a template to get started

Reviewer: ```suggestion
* copies from an existing example project as a template to get started

We're excited to see what you come up with!
```

Bit of an abrupt ending here.

.. list-table::
   :header-rows: 1
   :widths: 20 20 40 20

   * - Topic
     - Framework
     - Links
     - Description
   * - Basic Sphinx
     - Sphinx
     - `[Git] <https://github.com/readthedocs-examples/example-sphinx-basic/>`__ `[Rendered] <https://example-sphinx-basic.readthedocs.io/en/latest/>`__
     - Basic example of using Sphinx
   * - Basic MkDocs
     - MkDocs
     - `[Git] <https://github.com/readthedocs-examples/example-mkdocs-basic/>`__ `[Rendered] <https://example-mkdocs-basic.readthedocs.io/en/latest/>`__
     - Basic example of using MkDocs

Real-life examples
------------------

.. image:: _static/images/awesome-list.svg
  :alt: Awesome List badge
  :target: https://github.com/readthedocs-examples/

We maintain an **Awesome List** where you can contribute new shiny examples of using Read the Docs. Please refer to the instructions on how to submit new entries on `Awesome Read the Docs Projects <https://github.com/readthedocs-examples/>`_.

Contributing an example projects
--------------------------------

We would love to add more examples that showcase features of Read the Docs or great tools or methods to build documentation projects.

> .. admonition:: We **require** that an example project:

Reviewer: This is an odd syntax we don't use anywhere else. I think we just do a normal list, without the `...` at the front.

* Want to bootstrap a new documentation project?
* Want to showcase your own solution?

> The following example projects show a rich variety of uses of Read the Docs. You can use them for inspiration, for learning and for recipies to start your own documentation projects. View the *rendered* version of each project and then head over to the Git source to see how it's done and reuse the code.

Reviewer: Just a reminder that most of our docs use semantic line breaks: https://sembr.org/

Author: Ah yeah, I haven't quite worked it into my muscle memory, hopefully it will get there, soon! Thanks!

   * - Topic
     - Framework
     - Links
     - Description
   * - Basic Sphinx
     - Sphinx
     - `[Git] <https://github.com/readthedocs-examples/example-sphinx-basic/>`__ `[Rendered] <https://example-sphinx-basic.readthedocs.io/en/latest/>`__

>      - Basic example of using Sphinx

Reviewer: These could have a bit more description. What is showcased in this verison? `basic example` doesn't tell me much.  

```suggestion
     - Sphinx example with versioning and Python doc autogeneration
```

If it doesn't work well w/ the tables, could be a good reason to break them out into sections.

Author: But don't you think that it's much more inviting and easy to read in a table?

We maintain an **Awesome List** where you can contribute new shiny examples of using Read the Docs. Please refer to the instructions on how to submit new entries on `Awesome Read the Docs Projects <https://github.com/readthedocs-examples/>`_.

> Contributing an example projects
> --------------------------------

Reviewer: ```suggestion
Contributing an example project
-------------------------------
```

Author: Thanks!
